### PR TITLE
Fix data selector for joins

### LINKED ISSUE #12059: [Regression] Can't select table for joins in notebook on master
If you try to select a table to join to, all you see is the database. Clicking the database doesn't do anything.

image

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Fixes #12059

I branched off of #12064 to avoid a merge conflict.

The DataSelector maintained a filtered list of steps that omitted skipped steps. It was breaking the usage in joins because we started in the schema step, but that step was filtered. Separate from that filtered list, we also auto-advanced through skipped steps, so I figured we didn't need the list anymore.

I added a cypress test since we shouldn't need to figure out that joins are totally broken by clicking in the UI.

### CODE REVIEW

**Comment by mazameli** [comment]
I just realized that the "Saved Questions" option here is missing, which means I can't join to a saved question like I used to be able to. Here's how things used to look in **0.33.7.2**:

image

This PR does fix the ability to pick a table, which is great, but there's also still a bug now where if I click back to go to the database list in the join picker, clicking on the database still doesn't do anything. 

can't-click-database

**Review by mazameli — DISMISSED** [review]
[See comment](https://github.com/metabase/metabase/pull/12066#issuecomment-595523017)

**Comment by flamber** [comment]
@paulrosenzweig I'm not seeing the same behavior as the video I sent you, but I haven't tested with databases with many schemas (like stats), since I don't have that locally. Feels a lot smoother :+1:

File: frontend/src/metabase-lib/lib/metadata/Database.js [hunk 1]
```diff
@@ -93,15 +93,6 @@ export default class Database extends Base {
 
   /** Returns a database containing only the saved questions from the same database, if any */
   savedQuestionsDatabase(): ?Database {
-    const database = this.metadata
-      .databasesList()
-      .find(db => db.is_saved_questions);
-    if (database) {
```
Comment (tlrobinson) [inline frontend/src/metabase-lib/lib/metadata/Database.js]: Why is this no longer relevant? This filtered out questions that weren't part of the same underlying database.

Comment (paulrosenzweig) [reply frontend/src/metabase-lib/lib/metadata/Database.js]: Whoops I totally missed that. Fixing now...

**Review by tlrobinson — APPROVED** [review]

**Pull request merged**